OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs

```diff
@@ -120,5 +120,24 @@ protected static bool IsRearming(Actor a)
 
 			return false;
 		}
+
+		protected static bool FullAmmo(IEnumerable<AmmoPool> ammoPools)
+		{
+			return ammoPools.All(x => x.HasFullAmmo);
```

Comment:
This means we always return `false` if any pool doesn't have full ammo, so it would be more performant to do
```
foreach (var ap in ammoPools)
	if (!ap.HasFullAmmo)
		return false;

return true;
```

Reply:
checkmark